sanic_jwt/response.py

```diff
@@ -0,0 +1,77 @@
+import importlib
+from sanic.response import json
+
+
+class Response:
+    @staticmethod
+    async def get_access_token_output(request, user, config):
+        access_token = await request.app.auth.get_access_token(user)
+
+        output = {config.access_token_name: access_token}
+
+        return access_token, output
+
+    @staticmethod
+    def get_token_reponse(request,
+                          access_token,
+                          output,
+                          config,
+                          refresh_token=None):
+        response = json(output)
+
+        if config.cookie_set:
+            key = config.cookie_access_token_name
+            response.cookies[key] = str(access_token, 'utf-8')
+            response.cookies[key]['domain'] = \
+                config.cookie_domain
+            response.cookies[key]['httponly'] = \
+                config.cookie_httponly
+
+            if refresh_token and \
+                    config.refresh_token_enabled:
+                key = config.cookie_refresh_token_name
+                response.cookies[key] = refresh_token
+                response.cookies[key]['domain'] = \
+                    config.cookie_domain
+                response.cookies[key]['httponly'] = \
+                    config.cookie_httponly
+
+        return response
+
+    @staticmethod
+    def extend_authenticate(request,
+                            user=None,
+                            access_token=None,
+                            refresh_token=None):
+        return {}
+
+    @staticmethod
+    def extend_retrieve_user(request, user=None, payload=None):
+        return {}
+
+    @staticmethod
+    def extend_verify(request, user=None, payload=None):
+        return {}
+
+    @staticmethod
+    def extend_refresh(request,
+                       user=None,
+                       access_token=None,
+                       refresh_token=None,
+                       purported_token=None,
+                       payload=None):
+        return {}
+
+    @staticmethod
+    def exception_response(request, exception):
+        return json({
+            'exception': str(exception)
+        }, status=exception.status_code)
+
+
+def make_response(r):
```

Comment:
So, this actually overrides `sanic_jwt.endpoints.response` if none set, making it part of the `sanic_jwt` package - thus, global. Why not make it as a factory inside a custom `Initialize`?  It works fine for the other classes (like `configuration` ... Example:

```python
class MyInitialize(Initialize):
    response_class = MyResponse
```

Reply:
I think I made it global because I wanted it to be accessible to the endpoints. But looking now, it really should be passed into BaseEndpoint the same way config is way attaching individual endpoints. alright  that's in. thanks. PS response_class is already there.

Reply:
Nice! I haven't noticed `response_class` in there, but that's good. I think it would have no problem being local, since you can reuse the class whenever you want ...